### PR TITLE
Fix build issue with types, fix test runner and upgrade packages

### PR DESCRIPTION
**Pull request by mesaugat**
Should have been multiple pull requests but anyways.

- Fix type issues with `.from(table)`
- Remove old school dependencies `camelize` and `snakeize` and replace them with `camelcase-keys` and `snakecase-keys` (Because these packages have better TypeScript support)
- Upgrade all dependencies to latest version
- Add `source-map-support` for `nyc`
- Remove `mocha.opts` in favor of `.mocharc.json`

### CODE REVIEW

**Comment by kabirbaidhya** [comment]
Build passed?

**Comment by kabirbaidhya** [comment]
We can also add a check for GitHubActions to do `yarn lint && yarn test && yarn build`

**Comment by mesaugat** [comment]
Yep, that is another PR. `yarn test:coverage` is not working at the moment.

**Comment by mesaugat** [comment]
The build passes. @kabirbaidhya I'll mark it as ready for review once `test:coverage` works.

File: packages/db-model/package.json [hunk 1]
```diff
@@ -55,34 +54,33 @@
...
   "authors": "Kabir Baidhya <kabirbaidhya@gmail.com>",
   "dependencies": {
-    "camelize": "^1.0.0",
+    "camelcase-keys": "^6.2.2",
     "debug": "4.1.1",
-    "knex": "0.20.11",
+    "knex": "0.20.14",
     "ramda": "^0.27.0",
-    "snakeize": "^0.1.0"
+    "snakecase-keys": "mesaugat/snakecase-keys#38efc7b"
```
Comment (mesaugat) [inline packages/db-model/package.json:61]: Until https://github.com/bendrucker/snakecase-keys/pull/31 gets merged.

**Review by kabirbaidhya — APPROVED** [review]

**Pull request merged**